### PR TITLE
MES-3064 - Analytics for Rekey

### PR DESCRIPTION
**Pull request by TomBillingtonUK**
## Description and relevant Jira numbers
- Added event when you search for a test booked in rekey search
- Added an event which fires when you click rekey on a test result from rekey search
- When a test is being rekeyed analytics events are prefixed with Rekey when setting the current page a user is on. (Same as practice mode)

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

**Review by roberthall12 — APPROVED** [review]

**Review by rossfauldsbjss — APPROVED** [review]
looks good

File: src/shared/helpers/format-analytics-text.ts [hunk 1]
```diff
@@ -9,5 +9,8 @@ export function formatAnalyticsText(eventText: string, tests: TestsModel): strin
   if (isTestReportPracticeTest(tests)) {
     return `${AnalyticsEventCategories.PRACTICE_TEST} - ${eventText}`;
   }
+  if (getCurrentTest(tests).rekey) {
+    return `${AnalyticsEventCategories.REKEY} - ${eventText}`;
+  }
```
Comment (buzzub) [inline src/shared/helpers/format-analytics-text.ts:14]: Maybe add a quick test for this?

Comment (TomBillingtonUK) [reply src/shared/helpers/format-analytics-text.ts:14]: will add :)

**Review by buzzub — APPROVED** [review]
Good. Could maybe add another test for the `formatAnalyticsText` function

**Pull request merged**